src/main/java/uk/gov/register/filters/StripTrailingSlashRedirectFilter.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.register.filters;
+
+import com.google.common.net.HttpHeaders;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class StripTrailingSlashRedirectFilter implements javax.servlet.Filter {
+
+    private static String removeLastChar(String str) {
+        return str.substring(0, str.length() - 1);
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) { }
+
+    @Override
+    public void destroy() { }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest)request;
+
+        String path = httpRequest.getRequestURI();
+        if (path.endsWith("/") && path.length() > 1) {
+        int serverPort = httpRequest.getServerPort();
+            StringBuilder location = new StringBuilder();
```

Comment:
I would've thought there were a standard URL constructor/builder so we wouldn't have to have to construct it ourselves :(

Reply:
good point, I have switched to use a URIBuilder now.